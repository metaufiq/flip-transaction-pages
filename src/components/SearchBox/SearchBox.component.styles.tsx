import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 1,
    paddingHorizontal: 10,
    paddingVertical: 20,
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
})

export default styles;