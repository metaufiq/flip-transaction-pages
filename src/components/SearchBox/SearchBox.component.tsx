import React from "react";
import { TextInput, View } from "react-native";
import SelectInput from "../SelectInput";

import styles from "./SearchBox.component.styles";
import { Props } from "./SearchBox.component.types";

const SearchBox = ({placeholder, onChangeSearch, sortLabel, onChangeSort, options}: Props)=>{
  return(
    <View style={styles.container}>
      <TextInput 
        placeholder={placeholder}
        onChangeText={onChangeSearch}
      />
      <SelectInput
        label={sortLabel}
        onChange={onChangeSort}
        options={options}
      />
    </View>
  )
}

export default SearchBox;