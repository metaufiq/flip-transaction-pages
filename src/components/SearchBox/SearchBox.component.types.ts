import { Option } from "../SelectInput/SelectInput.component.types";

export interface Props{
  placeholder?: string,
  onChangeSearch: (value:string)=>void,
  options: Option[],
  sortLabel: string,
  onChangeSort: (value:string)=>void,
  value?: string
}