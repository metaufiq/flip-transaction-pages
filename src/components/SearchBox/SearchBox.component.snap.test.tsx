import { assertSnapshots } from "../../../test";
import SearchBox from ".";
import { Props } from "./SearchBox.component.types";

describe('SearchBoxSnapshot', () => {

  const mockProps:Props = {
    onChangeSort:()=>{},
    onChangeSearch:()=>{},
    sortLabel: 'label',
    options: [{
      label: 'label',
      value: 'value'
    }]
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render search input'
    },
    {
      props: {...mockProps, placeholder: 'placeholder'},
      desc: 'should render search input with placeholder'
    },
  ];

  assertSnapshots(SearchBox, configs);
});
