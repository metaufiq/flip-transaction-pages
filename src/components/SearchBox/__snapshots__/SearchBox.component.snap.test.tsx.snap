// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxSnapshot should render search input 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#fd6542",
        }
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`SearchBoxSnapshot should render search input with placeholder 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="placeholder"
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#fd6542",
        }
      }
    >
      label
    </Text>
  </View>
</View>
`;
